src/Http/Controllers/Admin/FeaturedController.php

```diff
@@ -5,35 +5,88 @@
 use A17\Twill\Models\Feature;
 use A17\Twill\Repositories\Behaviors\HandleMedias;
 use A17\Twill\Repositories\Behaviors\HandleTranslations;
-use DB;
-use Event;
+use Illuminate\Config\Repository as Config;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Database\DatabaseManager as DB;
+use Illuminate\Http\Request;
+use Illuminate\Routing\UrlGenerator;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+use Illuminate\View\Factory as ViewFactory;
 
 class FeaturedController extends Controller
 {
-    public function index()
+    /**
+     * @var DB
+     */
+    protected $db;
+
+    /**
+     * @var UrlGenerator
+     */
+    protected $urlGenerator;
+
+    /**
+     * @var Application
+     */
+    protected $app;
+
+    /**
+     * @var ViewFactory
+     */
+    protected $viewFactory;
+
+    /**
+     * @param DB $db
+     * @param UrlGenerator $urlGenerator
+     * @param Application $app
+     * @param ViewFactory $viewFactory
+     * @param Config $config
+     */
+    public function __construct(
+        DB $db,
+        UrlGenerator $urlGenerator,
+        Application $app,
+        ViewFactory $viewFactory,
+        Config $config
+    ) {
+        parent::__construct($app, $config);
+
+        $this->db = $db;
+        $this->urlGenerator = $urlGenerator;
+        $this->app = $app;
+        $this->viewFactory = $viewFactory;
+    }
+
+    /**
+     * @param Request $request
+     * @return array|\Illuminate\Contracts\View\View
+     */
+    public function index(Request $request)
     {
-        $featuredSectionKey = request()->segment(count(request()->segments()));
-        $featuredSection = config("twill.buckets.$featuredSectionKey");
-        $filters = json_decode(request()->get('filter'), true) ?? [];
+        $featuredSectionKey = $request->segment(count($request->segments()));
+        $featuredSection = $this->config->get("twill.buckets.$featuredSectionKey");
+        $filters = json_decode($request->get('filter'), true) ?? [];
 
-        $featuredSources = $this->getFeaturedSources($featuredSection, $filters['search'] ?? '');
+        $featuredSources = $this->getFeaturedSources($request, $featuredSection, $filters['search'] ?? '');
 
-        $contentTypes = collect($featuredSources)->map(function ($source, $sourceKey) {
+        $contentTypes = Collection::make($featuredSources)->map(function ($source, $sourceKey) {
             return [
                 'label' => $source['name'],
                 'value' => $sourceKey,
             ];
         })->values()->toArray();
 
-        if (request()->has('content_type')) {
-            $source = array_first($featuredSources, function ($source, $sourceKey) {
-                return $sourceKey == request('content_type');
+        if ($request->has('content_type')) {
+            $source = Arr::first($featuredSources, function ($source, $sourceKey) use ($request) {
```

Comment:
The only thing compared is the index key. So `Arr::first` isn't required here.
```suggestion
            $source = $featuredSources[$request->get('content_type')];
```

Reply:
You are right.